example/netcdf/t8_write_forest_netcdf.cxx

```diff
@@ -13,6 +13,99 @@
 
 T8_EXTERN_C_BEGIN ();
 
+/* Function that times the duration of writing out the netCDF File, given a specific variable storage and access pattern */
+static void
+t8_example_time_netcdf_writing_operation (t8_forest_t forest,
+                                          sc_MPI_Comm comm,
+                                          int netcdf_var_storage_mode,
+                                          int netcdf_var_mpi_access,
+                                          const char *title)
+{
+#if T8_WITH_NETCDF
+  double              start_time, end_time, duration, global;
+  int                 retval;
+  sc_MPI_Barrier (comm);
+  start_time = sc_MPI_Wtime ();
+  t8_forest_write_netcdf_ext (forest, title,
+                              "Performance Test: uniformly refined Forest", 3,
+                              0, NULL, comm, netcdf_var_storage_mode,
+                              netcdf_var_mpi_access);
+
+  sc_MPI_Barrier (comm);
+  end_time = sc_MPI_Wtime ();
+  duration = end_time - start_time;
+  retval =
+    sc_MPI_Reduce (&duration, &global, 1, sc_MPI_DOUBLE, sc_MPI_MAX, 0, comm);
+  SC_CHECK_MPI (retval);
+  t8_global_productionf
+    ("The time elapsed to write the netCDF-4 File is: %f\n\n", global);
+#endif
+}
+
+/* Function that stores the given forest in a netCDF-4 File using the different netCDF variable storage and mpi-access patterns */
+void
+t8_example_compare_performance_netcdf_var_properties (sc_MPI_Comm comm,
+                                                      int
+                                                      forest_refinement_level)
+{
+#if T8_WITH_NETCDF
+  t8_cmesh_t          cmesh;
+  t8_forest_t         forest;
+  t8_scheme_cxx_t    *default_scheme;
+
+  /* Create a default scheme */
+  default_scheme = t8_scheme_new_default_cxx ();
+
+  /* Construct a 3D hybrid hypercube as a cmesh */
+  cmesh = t8_cmesh_new_hypercube_hybrid (3, comm, 0, 0);
+
+  /* Build a (partioined) uniform forest */
+  forest =
+    t8_forest_new_uniform (cmesh, default_scheme, forest_refinement_level, 0,
+                           comm);
```

Comment:
Possible extension for the future: Adapt the forest before writing so that we do not always write a uniform forest.